client/ui/components/AddonForm.jsx

```diff
@@ -56,33 +52,31 @@ class AddonForm extends React.Component {
     }
   }
 
-  componentWillUnmount() {
-    this.props.resetAll();
-  }
-
   render() {
-    const {
-      activeAddon, createAddon, handleSubmit, save, saveAndContinue, updateAddon,
-    } = this.props;
-    const isSaving = createAddon.loading || updateAddon.loading;
-    const saveError = createAddon.error || updateAddon.error;
+    const { addon, fetchRequest, handleSubmit, pk, saveAddon, saveRequest } = this.props;
+    const isSaving = saveRequest.loading;
```

Comment:
I'm fine with this, but I think in the future we should consider if we can return the Promise for fetching the addons to redux-form such that we can use the redux-form `submitting` and error messaging stuff.